File: com/white/RemoveElement.java
package com.white;

public class RemoveElement {

    // 27. Remove Element
    // needs further investigation
    public static int removeElement(int[] nums, int val) {
        if (nums==null || nums.length==0) {
            return 0;
        }

        int index = 0;
        int currFetchIndex = nums.length;
        while (index<currFetchIndex) {
            if (nums[index]==val) {
                nums[index] = nums[currFetchIndex-1];
                currFetchIndex--;
            } else {
                index++;
            }
        }

        return currFetchIndex;
    }
}

File: LeetcodeGo/RemoveElement.go
package main

// 27. Remove Element
func removeElement(nums []int, val int) int {
	index1 := 0
	index2 := len(nums) - 1

	for index1 <= index2 {
		if nums[index1] == val {
			swap(nums, index1, index2)
			index2--
		} else if nums[index1] != val {
			index1++
		}
	}

	return index1
}

func swap(nums []int, a int, b int) {
	temp := nums[a]
	nums[a] = nums[b]
	nums[b] = temp
}
